pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1692,7 +1692,18 @@ class Timestamp(_Timestamp):
             )
             # Once this deprecation is enforced, we can do
             #  return Timestamp(ts_input).tz_localize(tzobj)
-        ts = convert_to_tsobject(ts_input, tzobj, unit, 0, 0, nanosecond or 0)
+        if nanosecond is None:
+            nanosecond = 0
+        elif not (999 >= nanosecond >= 0):
+            warnings.warn(
```

Comment:
could we just fix these now?  The -1 case at least looks very broken

Reply:
Hmm probably could be okay for 2.0 API change. Technically 999 < nanosecond < int32_max works okay, but sure can fix it in one go.